g2p/apply.py

```diff
@@ -56,20 +61,61 @@ def __init__(
         self.variants_number = variants_number
         self.word_list = word_list_file
         self.filter_empty_words = filter_empty_words
+        self.concurrent = concurrent
+
+        self.out_g2p_lexicon_parts = None
+        self.out_g2p_untranslated_parts = None
+        self.word_list_parts = None
 
         self.out_g2p_lexicon = self.output_path("g2p.lexicon")
         self.out_g2p_untranslated = self.output_path("g2p.untranslated")
 
+        if self.concurrent > 1:
+            self.out_g2p_lexicon_parts = [
+                self.output_path(f"g2p.lexicon.{i}")
+                for i in range(1, self.concurrent + 1)
+            ]
+            self.out_g2p_untranslated_parts = [
+                self.output_path(f"g2p.untranslated.{i}")
+                for i in range(1, self.concurrent + 1)
+            ]
+            num_digits = len(str(self.concurrent))
+            self.word_list_parts = [
+                f"words.{i:0{num_digits}d}" for i in range(1, self.concurrent + 1)
+            ]
+
         self.rqmt = {"cpu": 1, "mem": 1, "time": 2}
 
     def tasks(self):
-        yield Task("run", rqmt=self.rqmt)
+        if self.concurrent == 1:
+            yield Task("run", rqmt=self.rqmt)
+        else:
+            yield Task("create_files", mini_task=True)
+            yield Task(
+                "run", rqmt=self.rqmt, args=range(1, self.concurrent + 1)
+            )
+            yield Task("merge", mini_task=True)
```

Comment:
do we need special handling for the case `concurrent==1`? Sure the split and merge tasks would be no-ops then, but in principle you could treat `concurrent==1` the same as "`concurrent==N` with N=1" (also in terms of filenames/logic within the run task)

This would make it more general / cleaner to add more functionality later.

Reply:
I generalized the code to exclude the special case handling of concurrent=1.